lib/omniauth/smart/conformance.rb

```diff
@@ -66,14 +68,16 @@ def self.conformance_uri_for_server(server_uri)
       end
 
       def is_known_security_extension?(security_extension_json)
-        security_extension_json["url"] == "http://fhir-registry.smarthealthit.org/StructureDefinition/oauth-uris"
+        security_extension_json["url"] == KNOWN_SECURITY_EXTENSION_URL
       end
 
       def parse_security_extension
         if @conformance_json.has_key?("Conformance")
           @conformance_json["Conformance"]["rest"]["security"]["extension"]
         else
-          @conformance_json["rest"][0]["security"]["extension"][0]
+          @conformance_json["rest"][0]["security"]["extension"].detect do |extension|
+            extension["url"] == KNOWN_SECURITY_EXTENSION_URL
+          end
```

Comment:
```suggestion
          @conformance_json["rest"][0]["security"]["extension"].find { |e| is_known_security_extension?(e) }
```

find vs detect doesn't matter too much but I think it's conveys that we're looking for a specific extension.